### PR TITLE
Feature/unified connector new setup

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Setup connectors setup page in project panel

- Depends on https://github.com/the-deep/server/tree/feature/new-unified-connector
- Depends on https://github.com/the-deep/deeper/pull/974

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/ProjectEdit/Connector/schema.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            source: [requiredCondition],
+            // FIXME: Define better schema
+            params: [],
+        };
+        const reliefWebParamsSchema = {
+            fromDate: [],
+            toDate: [],
+            primaryCountry: [requiredCondition],
+            country: [],
+        };
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/schema.ts]: Why not just define this directly on line 47

File: app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/ReliefWebParamsInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    const {
+        name,
+        value,
+        error: riskyError,
+        onChange,
+    } = props;
+
+    const error = getErrorObject(riskyError);
+    // FIXME: Look into if default values is good
+    const setParamsFieldValue = useFormObject(name, onChange, value ?? {});
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/ReliefWebParamsInput/index.tsx]: Let's properly define the default value.

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+import styles from './styles.css';
+
+type SupportedSource = 'RELIEF_WEB' | 'UNHCR';
+const supportedSources = ['RELIEF_WEB', 'UNHCR'];
+
+const sourcesLabel = {
+    RELIEF_WEB: 'Relief Web',
+    UNHCR: 'UNHCR',
+};
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx]: Let's use a stricter type

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+    const error = getErrorObject(riskyError);
+
+    const handleSourceClick = useCallback((key: SupportedSource) => {
+        setFieldValue(
+            (oldSources: PartialSourceType[] | undefined = []) => {
+                const selectedSourceIndex = oldSources
+                    ?.findIndex((source) => source.source === key);
+
+                if (selectedSourceIndex === -1) {
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx:151]: What if selectedSourceIndex is undefined?

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+            },
+            'sources' as const,
+        );
+    }, [setFieldValue]);
+
+    const usedSourceMap = listToMap(
+        value?.sources?.filter((source) => isDefined(source.source)),
+        (source) => (source.source),
+        () => true,
+    );
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx]: Let's memoize this

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+        validate,
+        createConnector,
+    ]);
+
+    return (
+        <Modal
+            className={_cs(styles.editConnectorModal, className)}
+            onCloseButtonClick={onClose}
+            size="cover"
+            heading={connectorId ?? 'Add New Connector'}
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx]: Huh?

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+            footerActions={(
+                <Button
+                    name={undefined}
+                    onClick={handleSubmit}
+                    disabled={pristine}
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx]: Let's disable this button request is in flight

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+    const connectorSourceRendererParams = useCallback((
+        key: string,
+        data: PartialSourceType,
+        index,
+    ) => ({
+        name: index,
+        error: sourcesError?.[key],
+        value: data,
+        onChange: onRowChange,
+    }), [onRowChange, sourcesError]);
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx]: Let's pass disabled

File: app/views/ProjectEdit/Connector/index.tsx [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+
+interface ItemProps {
+    itemKey: string;
+    title: string;
+    createdAt: string;
+    onClick: (key: string) => void;
+    isSelected: boolean;
+}
+
+function Item(props: ItemProps) {
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/index.tsx]: Item is too generic

File: app/views/ProjectEdit/Connector/index.tsx [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+        isSelected,
+    } = props;
+
+    const handleItemClick = useCallback(() => {
+        onClick(itemKey);
+    }, [onClick, itemKey]);
+
+    return (
+        <RawButton
+            name={`item-${itemKey}`}
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/index.tsx]: Why not just pass the actual itemKey and skip using handleItemClick?

File: app/views/ProjectEdit/Connector/index.tsx [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+    const [connectorList, setConnectorList] = useState<ConnectorMini[]>([]);
+    const [connectorsCount, setConnectorsCount] = useState<number>(0);
+    const [offset, setOffset] = useState<number>(0);
+
+    const connectorsRendererParams = useCallback((key: string, data: ConnectorMini) => ({
+        itemKey: key,
+        title: data.title,
+        createdAt: data.createdAt,
+        onClick: setSelectedConnector,
+        isSelected: key === selectedConnector,
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/index.tsx]:
```suggestion
        selected: key === selectedConnector,
```

File: app/views/ProjectEdit/Connector/index.tsx [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+                page: offset / PAGE_SIZE + 1,
+            },
+            onCompleted: (response) => {
+                const unifiedConnectors = response?.project?.unifiedConnector?.unifiedConnectors;
+                setConnectorsCount(unifiedConnectors?.totalCount ?? 0);
+
+                setConnectorList((oldConnectors) => ([
+                    ...oldConnectors,
+                    ...(unifiedConnectors?.results ?? []),
+                ]));
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/index.tsx]: Why not just read this directly?

File: app/views/ProjectEdit/Connector/index.tsx [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+        <div className={_cs(className, styles.connector)}>
+            <Container
+                className={styles.leftContainer}
+                contentClassName={styles.content}
+            >
+                <ListView
+                    className={styles.connectorList}
+                    pending={connectorsGetPending}
+                    errored={false}
+                    filtered={false}
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/index.tsx]: We should handle this

File: app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/index.tsx [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+import {
+    PartialSourceType,
+} from '../../schema';
+import ReliefWebParamsInput from './ReliefWebParamsInput';
+
+import styles from './styles.css';
+
+interface Props {
+    className?: string;
+    name: number | undefined;
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/index.tsx]: Does this work?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx (hunk 6 above)
Comment (AdityaKhatri) [reply app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx]: This is in progress. I'll add a note

File: app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/index.tsx (hunk 14 above)
Comment (AdityaKhatri) [reply app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/index.tsx]: yes

File: app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/ReliefWebParamsInput/index.tsx [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+        value,
+        error: riskyError,
+        onChange,
+    } = props;
+
+    const error = getErrorObject(riskyError);
+    const setParamsFieldValue = useFormObject(name, onChange, value);
+
+    return (
+        <>
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/ReliefWebParamsInput/index.tsx:48]: Let's handle NonFieldError

File: app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx (hunk 6 above)
Comment (tnagorra) [reply app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx]: Let's just have a "Edit" or "New" connector title

File: app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/ReliefWebParamsInput/index.tsx (hunk 15 above)
Comment (AdityaKhatri) [reply app/views/ProjectEdit/Connector/EditConnectorModal/ConnectorSourceForm/ReliefWebParamsInput/index.tsx:48]: done

File: app/views/ProjectEdit/Connector/schema.ts [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+const sourcesSchema: SourcesSchema = {
+    keySelector: (col) => col.clientId,
+    member: (): SourcesSchemaMember => sourceSchema,
+};
+
+export const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        clientId: [requiredStringCondition],
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/schema.ts]: Let's remove this condition

File: app/views/ProjectEdit/Connector/schema.ts [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+    ArraySchema,
+    PartialForm,
+    defaultUndefinedType,
+    requiredStringCondition,
+    requiredCondition,
+} from '@togglecorp/toggle-form';
+
+import { ConnectorInputType } from './types';
+
+export type PartialFormType = PartialForm<ConnectorInputType, 'clientId' | 'source' | 'params'>;
```
Comment (tnagorra) [inline app/views/ProjectEdit/Connector/schema.ts]: Why add "params" on PartialForm?

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**